storage: add storage.tombstone_dense_compaction_threshold setting

RaduBerinde opened the pull request:
Add a cluster setting to control the tombstone-dense compaction
threshold.

Fixes: #155715
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/155739)
<!-- Reviewable:end -->

RaduBerinde committed:
storage: add storage.tombstone_dense_compaction_threshold setting

Add a cluster setting to control the tombstone-dense compaction
threshold.

Fixes: #155715
Release note: None

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @jbowens reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/155739#-:-OcBU1sjCd9VdNeocjyl:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde)



<!-- Sent from Reviewable.io -->
